test/unit/org/apache/cassandra/index/sai/cql/VectorCompactionTest.java

```diff
@@ -142,6 +179,190 @@ public void testZeroOrOneToManyCompaction()
+
+            assertFalse(i.areSegmentsLoaded());
+            assertEquals(0, i.indexFileCacheSize());
+            assertThrows(UnsupportedOperationException.class, i::getSegments);
+            // This is vector specific, so if we broaden the functionality of V1MetadataOnlySearchableIndex, we might
+            // need to update this.
+            assertArrayEquals(new byte[] { 0, 0, 0, 0 }, ByteBufferUtil.getArray(i.minTerm()));
+            assertArrayEquals(new byte[] { 0, 0, 0, 0 }, ByteBufferUtil.getArray(i.maxTerm()));
+            // Confirm count matches sstable row id range (the range is inclusive)
+            var sstableDiff = i.maxSSTableRowId() - i.minSSTableRowId() + 1;
+            assertEquals(i.getRowCount(), sstableDiff);
+
+            var postingsStructures = i.getPostingsStructures().collect(Collectors.toList());
+            assertEquals(1, postingsStructures.size());
+            assertEquals(V5VectorPostingsWriter.Structure.ONE_TO_ONE, postingsStructures.get(0));
+        });
+
+        // Now compact (this exercises the code, but doesn't actually prove that we used the past segment's PQ)
+        compact();
+
+        // Reload and make queryable
+        CassandraRelevantProperties.SAI_INDEX_READS_DISABLED.setBoolean(false);
+        reloadSSTableIndex();
+
+        // Assert that we have a product quantization
+        var index = (StorageAttachedIndex) getIndexManager(keyspace(), indexName).listIndexes().iterator().next();
+        var sstableIndexes = index.getIndexContext().getView().getIndexes();
+        assertEquals(1, sstableIndexes.size());
+        var searcher = (V2VectorIndexSearcher) sstableIndexes.iterator().next().getSegments().iterator().next().getIndexSearcher();
+        // We have a PQ even though we have fewer than MIN_PQ_ROWS because we used the original seed.
+        // We only test like this to confirm that it worked, not because this is a strictly required behavior.
+        // Further, it's unlikely we'll have so few rows.
+        assertEquals(VectorCompression.CompressionType.PRODUCT_QUANTIZATION, searcher.getCompression().type);
+        assertEquals(V5VectorPostingsWriter.Structure.ONE_TO_ONE, searcher.getPostingsStructure());
+
+        // Confirm we can query the data
+        var result = execute( "SELECT * FROM %s ORDER BY v ANN OF ? LIMIT 5", randomVectorBoxed(2));
+        assertEquals(5, result.size());
+    }
+
+    @SuppressWarnings("resource")
+    @Test
+    public void testSaiVectorIndexPostingStructure()
+    {
+        createTable("CREATE TABLE %s (pk int, v vector<float, 2>, PRIMARY KEY(pk))");
+        var indexName = createIndex("CREATE CUSTOM INDEX ON %s(v) USING 'StorageAttachedIndex'");
+
+        // We'll manage compaction manually for better control
+        disableCompaction();
+
+        // first sstable has one-to-one
+        for (int i = 0; i < MIN_PQ_ROWS; i++)
+            execute("INSERT INTO %s (pk, v) VALUES (?, ?)", i, randomVectorBoxed(2));
```

Comment:
side comment (not asking for a change): 
make randomVectorBoxed with only 2 dimensions could return duplicate vectors and make the test flaky